src/parquet.cc

```diff
@@ -48,7 +47,32 @@ const uint8_t* InMemoryInputStream::Read(int num_to_read, int* num_bytes) {
   return result;
 }
 
+ScopedInMemoryInputStream::ScopedInMemoryInputStream(int64_t len) {
+  buffer_.resize(len);
+  stream_.reset(new InMemoryInputStream(buffer_.data(), buffer_.size()));
+}
+
+uint8_t* ScopedInMemoryInputStream::data() {
+  return buffer_.data();
+}
+
+int64_t ScopedInMemoryInputStream::size() {
+  return buffer_.size();
+}
+
+const uint8_t* ScopedInMemoryInputStream::Peek(int num_to_peek,
+                                               int* num_bytes) {
+  return stream_->Peek(num_to_peek, num_bytes);
+}
+
+const uint8_t* ScopedInMemoryInputStream::Read(int num_to_read,
+                                               int* num_bytes) {
+  return stream_->Read(num_to_read, num_bytes);
+}
+
+
 ColumnReader::~ColumnReader() {
+  delete stream_;
```

Comment:
This has unclear ownership semantics. We should make this a unique ptr too. Take a look at this example
http://en.cppreference.com/w/cpp/memory/unique_ptr